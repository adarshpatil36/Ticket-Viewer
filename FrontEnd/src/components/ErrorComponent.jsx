import React from "react";

export default function ErrorComponent({ message }) {
  return (
    <div className="ErrorComponent">
      Oops, there is an error : <span> {message}</span>
    </div>
  );
}
