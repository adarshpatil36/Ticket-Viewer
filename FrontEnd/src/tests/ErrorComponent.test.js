import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ErrorComponent from "../components/ErrorComponent";

describe("ErrorComponent testing", () => {
  const msg = "Network error";
  it("renders TicketItem component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ErrorComponent message={msg} />, div);
  });

  it("renders TicketItem with proper label", () => {
    const { getByTestId } = render(<ErrorComponent message={msg} />);
    expect(getByTestId("errorComponent")).toHaveTextContent(
      "Oops, there is an error : Network error"
    );
  });

  it("renders correctly", () => {
    const tree = renderer.create(<ErrorComponent message={msg} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
