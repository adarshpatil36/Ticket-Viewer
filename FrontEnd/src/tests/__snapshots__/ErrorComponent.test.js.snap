// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent testing renders correctly 1`] = `
<div
  className="ErrorComponent"
  data-testid="errorComponent"
>
  Oops, there is an error : 
   
  <span>
     
    Network error
  </span>
</div>
`;
